### PR TITLE
ACL token support for all p2 consul-interacting binaries

### PR DESCRIPTION
**Pull request by tummychow**
mostly self-explanatory

### CODE REVIEW

File: bin/p2-bootstrap/bootstrap.go [hunk 1]
```diff
@@ -102,7 +103,9 @@ func VerifyConsulUp(timeout string) error {
 		return nil
 	}
 
-	store := kp.NewStore(kp.Options{})
+	store := kp.NewStore(kp.Options{
+		Token: *consulToken, // not actually necessary because this endpoint is unauthenticated
```
Comment (anthonybishopric) [inline bin/p2-bootstrap/bootstrap.go]: The comment isn't necessarily true - different organizations may point to an existing consul setup with strict reader ACLs.

Comment (tummychow) [inline bin/p2-bootstrap/bootstrap.go]: No the catalog endpoints actually cannot be authenticated, consul doesn't have that feature. kp contains primarily kv endpoints (which support ACLs) but it also contains service endpoints for service registration (no ACL support) and catalog endpoints for pinging (no ACL support).

**Comment by tummychow** [comment]
Took it for a spin in universe (with a consul cluster that defaulted to deny) and it works fine. As we discussed, bootstrapping a new node is still a bit of an open question (have to generate an ACL for its permissions and pass it to the preparer somehow) but we don't need to implement that in p2 itself.

**Comment by anthonybishopric** [comment]
The wiki page for setting up a cluster should include a note about this, since pgp is not currently mentioned anywhere on there and will surely screw up anyone who tries to set up a cluster now. As long as the documentation makes it clear how to bootstrap a preparer with the `consul_token_path`, this doesn't need to be solved by p2 directly. :+1:

**Comment by tummychow** [comment]
I added PGP notes to the wiki. ACLs are optional because consul defaults to allowing all un-ACLed requests, so new users shouldn't have to worry about this functionality.

**Pull request merged**